src/gui/guiFormSpecMenu.cpp

```diff
@@ -1256,15 +1381,33 @@ void GUIFormSpecMenu::parseVertLabel(parserData* data, const std::string &elemen
 
 		MY_CHECKPOS("vertlabel",1);
 
-		v2s32 pos = getElementBasePos(false, &v_pos);
+		v2s32 pos;
+		core::rect<s32> rect;
 
-		core::rect<s32> rect = core::rect<s32>(
-				pos.X, pos.Y+((imgsize.Y/2)- m_btn_height),
+		if (data->real_coordinates) {
+			pos = getRealCoordinateBasePos(false, &v_pos);
+
+			// Add offset for smaller rect.
+			pos.X += (imgsize.X / 4);
+
+			rect = core::rect<s32>(pos.X, pos.Y,
+				pos.X + imgsize.Y / 2,
+				pos.Y + font_line_height(m_font) *
+				(text.length() + 1));
```

Comment:
> ```
> 			// Actually text.length() would be correct but adding +1 avoids
> 			// breaking mods.
> ```

It might make sense adding this comment here too or not adding `1`.

Reply:
Actually, `text.length()` without adding 1 makes the `vertlabel` too short and cuts off the text, so that comment is actually incorrect.  I should probably remove that completely.

Reply:
You could also comment that the `+ 1` is for not cutting off the label.